### PR TITLE
Add an option for classic search

### LINKED ISSUE #3143: New Search Engine should link to right branch (stable/main/preview pr branch)
The search feature should match the branch that the docs loaded in. Why? The use case I often have is I use the search bar to quickly navigate to the page I had just edited in my PR to see how it'd render in prod. The new search engine produces results that always directs to stable, though, so there's no easy way to navigate to the page I wanted to check. This is a regression from the previous search experience.

For example, in the following preview, I'm looking for the custom operators page, which I had modified in the PR. I search for it in the preview docs, but all the results point to stable. Ideally, these would point to the docs built for my PR branch, which was the old behavior.

image

It would also be good for those look at docs on main to stay in docs on main (vs be redirected to stable).

## Alternatives

Allow the old search engine

### PR DESCRIPTION
**Pull request by svekars**
Fixes https://github.com/pytorch/tutorials/issues/3143 

cc @brycebortree @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: docs/source/_templates/layout.html [hunk 1]
```diff
@@ -28,10 +28,71 @@
...
+      const googleSearchBox = document.getElementById('googleSearchBox');
+      const sphinxSearchBox = document.getElementById('sphinxSearchBox');
+      // Function to toggle search box visibility
+      function toggleSearchBox(searchType) {
+        googleSearchBox.style.display = searchType === 'google' ? 'block' : 'none';
+        sphinxSearchBox.style.display = searchType === 'sphinx' ? 'block' : 'none';
+      }
+      // Determine the default search type
+      let defaultSearchType = 'google';
+      if (window.location.href.startsWith('https://docs-preview.pytorch.org/')) {
```
Comment (albanD) [inline docs/source/_templates/layout.html:69]: Should this also be the default for non-stable version? As google only links to the stable version?

Comment (svekars) [reply docs/source/_templates/layout.html]: I changed this to fall back to sphinx for everything but stable branch.

**Review by albanD — APPROVED** [review]
SGTM!

**Pull request closed without merging**